### PR TITLE
Feature/edit profile screen

### PR DESCRIPTION
**Pull request by sandroquist**
This pull request implement the edit profile screen 
- Show a profile picture with a button for editing
- Show a username with edit and clear options and limited to 20 characters
- Show a description with edit and clear options and limited to 100 characters
- Show a save button
- Implement screen accessibility from the profile screen and a go back arrow
- Implement the associated tests

### CODE REVIEW

**Review by mikhailperevoznyk — APPROVED** [review]
The overall implementation of the edit screen is good. The only thing that I would personally add is some different hardcoded profile pictures so that you could have implemented a function that edits the profile screen by choosing one of the different profile pictures. But of course it's something that wasn't ask from you so it's only a personal detail that I have found. Other than that everything looks good.

File: app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                        colors =
+                            ButtonDefaults.buttonColors(containerColor = lightThemeBackground)) {
+                          Text(
+                              text = "Save",
+                              fontWeight = FontWeight.Bold,
+                              fontSize = 16.sp,
+                              color = MaterialTheme.colorScheme.primary)
+                        }
+                  }
+            }
```
Comment (memagnin) [inline app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt:164]: We have a reusable **PrincipalButton** composable in the **MainComponents** file, designed for this type of commonly used button in the app. Consider using it here to reduce code duplication.

File: app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            },
+            navigationIcon = {
+              CornerIcons(
+                  onClick = { navigationActions.goBack() },
+                  icon = Icons.AutoMirrored.Filled.ArrowBack,
+                  contentDescription = "GoBack",
+                  modifier = Modifier.testTag("editProfileBackButton"),
+              )
+            })
+      },
```
Comment (memagnin) [inline app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt:94]: We have a reusable **ScreenTopAppBar** composable in the **MainComponents** file, designed for this type of commonly used top app bar in the app. Consider using it here to reduce code duplication.

File: app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+          Modifier.size(32.dp)
+              .background(color = backgroundColor, shape = CircleShape)
+              .clickable { /* Handle click */}) {
+        Icon(
+            imageVector = icon,
+            contentDescription = "Edit",
+            tint = LightGray,
+            modifier = Modifier.padding(6.dp))
+      }
+}
```
Comment (memagnin) [inline app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt:181]: I've noticed that this composable will be of use in a future screen, Playlist Overview Screen. Adding it to the MainComponents file might be a good idea.

**Review by memagnin — CHANGES_REQUESTED** [review]
Overall, this PR looks good. I’ve suggested a few changes where you could use general composables, from the MainComponents file, to reduce duplicated code.

**Review by memagnin — APPROVED** [review]
This PR is very well done! It effectively uses the common set of composables that are frequently reused across the app to construct a consistent and maintainable UI.

**Pull request merged**